src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

```diff
@@ -56,5 +56,9 @@
       <!-- Methods is used by VS Tasks Window. -->
       <method name="GetActiveTaskFromId" />
     </type>
+    <!-- Accessed via private reflection by an external tracing controller. -->
+    <type fullname="System.Diagnostics.Tracing.EventPipe*" />
+    <!-- The private Event methods are accessed by private reflection in the base EventSource class. -->
+    <type fullname="System.Diagnostics.Tracing.NativeRuntimeEventSource" />
```

Comment:
Where exactly is the dependency needed? We have more options to keep them only when needed

Reply:
This is taken from CoreCLR's IL linker descriptor so haven't looked into where all is needed in detail (except the comments above). For this specific test Linker eliminated IEventProvider.EventActivityIdControl and EventPipeEventProvider implementation of the same method, but since we now share most of the managed code with regards to EventListener/EventSource as well as EventPipe, thought it would make sense to use the same ILLinker rules for S.P.C.

Reply:
Found some examples of private reflection of System.Diagnostics.Tracing.EventPipe* in CoreCLR tests, runtime\src\tests\tracing\common\TraceConfiguration.cs.

Reply:
And for the test fixed and enabled in this PR it uses private reflection to get EventPipeEventProvider implementation of  IEventProvider.EventActivityIdControl:

runtime\src\tests\tracing\eventactivityidcontrol\EventActivityIdControl.cs

Reply:
Yep, it looks like this is actually needed only for testing in which case it should go to `ILLink.Descriptors_LibraryBuild.xml.xml` which @eerhardt is going to add soon.

Reply:
But should we still move that section as part of this PR into the shared section and then we could eliminate/move things only needed for testing after? I believe the NativeRuntimeEventSource needs to be included regardless (but checking the System.Diagnostics.Tracing.EventSource.IsSupported feature) since it is used at runtime to create the manifest,

Reply:
What about if we keep that in CoreCLR for now and move it to _LibraryBuild.xml in follow up PR

Reply:
And add rule to our Mono specific file as well, since we need for the tests enabled and fixed by this PR in order to pass.

Reply:
We don't want to keep this dependency on workloads we are shipping with net5. If you need it for the tests right now it needs to go to `ILLink.Descriptors_LibraryBuild.xml`

Reply:
OK, then this PR needs to wait on that file to appear and we can then move System.Diagnostics.Tracing.EventPipe* into ILLink.Descriptors_LibraryBuild.xml. For the NativeRuntimeEventSource we probably need to keep it in ILLink.Descriptors.Shared.xml since it is used at runtime to generate manifest/descriptors.

Reply:
My understanding of `NativeRuntimeEventSource` is that only CoreCLR actually logs these events. So putting this into Mono only to enable tests doesn't make sense to me. If an end-user's app doesn't actually get these events logged, then this class shouldn't exist in Mono.

cc @noahfalk @sywhang 

> If you need it for the tests right now it needs to go to ILLink.Descriptors_LibraryBuild.xml

I don't think either of these should move to a `_LibraryBuild.xml` file since a trimmed app could use them.

Reply:
In future Mono will also log these events through EventPipe but currently not enabled since have not hooked up native event emitting in Mono runtime.

Reply:
But until then, we can delay including that class, since it is not needed for this particular test to pass and move it when we start to emit the native runtime events from MonoVM.